lib/help_scout/api/access_token.rb

```diff
@@ -1,10 +1,40 @@
 # frozen_string_literal: true
 
+require 'date'
+
 module HelpScout
   class API
     class AccessToken
       class << self
         def create
+          cache.present? ? fetch_token : request_token
+        end
+
+        def refresh!
+          HelpScout.api.access_token = create
+        end
+
+        private
+
+        def cache
+          HelpScout.configuration.token_cache
+        end
+
+        def cache_key
+          HelpScout.configuration.token_cache_key
+        end
+
+        def fetch_token
```

Comment:
Solid 👌. I like that you're using the `expires_in` to actually drop the key in Redis.

Reply:
Yeah I'm hoping that pretty much any cache implementation will either accept the option or just silently ignore it (vs. blowing up because of an unrecognized arg etc.), so if you come across any cache implementations that don't, let me know :)